### DISCUSSION TITLE
How do I create and link a stand-alone page?

### DISCUSSION THREAD

**Discussion by storyteller-dev** [ctreffe/alfred#133] [Alfred Q&A] [answered]
I have a short debriefing at the end of my experiment, but I'd like to also include a longer version for those, who are interested in the topic.
I'd like to put a link in the text of the short version that participants can click to open a page with the longer version.
How can I create such a page outside the linear scope of the experiment and put a link in a text element?

**Comment by jobrachem** [comment, marked as answer]
I think your intention can be achieved easier with a collapsed `alfred3.Card`, but stand-alone pages are possible. I'll explain both ways.

## Stand-alone page

You can create a page that is hidden by default and add buttons for jumping.

In the example below, a click on the `alfred3.Button` on the first page triggers a function that will allow the hidden page to be shown. We also have a `alfred3.JumpButton` on the hidden page that takes participants back to the original page. To complete the setup, the hidden page will set its own `should_be_shown` attribute back to `False` each time it is hidden - this way, we ensure that our hidden page can only be accessed by a click on our button.

```python
import alfred3 as al

exp = al.Experiment()


@exp.member
class Demo(al.Page):
    title = "Demo Page"

    def on_exp_access(self):
        self += al.Button(
            "Jump to hidden page",
            func=self.show_hidden_page,
            followup="jump>HiddenPage",
            align="center",
        )

    def show_hidden_page(self):
        self.exp.HiddenPage.should_be_shown = True


@exp.member
class HiddenPage(al.Page):
    should_be_shown = False
    title = "Hidden page"

    def on_exp_access(self):
        self += al.JumpButtons(
            ("Back", "Demo"), # (displayed text, name of target page)
            name="jump_back",
            width="narrow",
            align="center",
            button_round_corners=False,
        )
        self += al.HideNavigation()

    def on_each_hide(self):
        self.should_be_shown = False

```

Why use a `alfred3.Button` on one page and a `alfred3.JumpButtons` element on the other page? The Button element allows us to call a Python function before jumping. The JumpButtons element is simply for jumping directly.

## Collapsed card

The `alfred3.Card` ([docs](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.element.display.Card.html?highlight=card#alfred3.element.display.Card)) is a great tool for displaying long text. Here's an example:

```python
import alfred3 as al
exp = al.Experiment()

@exp.member
class Demo(al.Page):
    
    def on_exp_access(self):
        self += al.Card(
            header="Card Header",
            body="**This text** is placed in the body.",
            collapse=True
        )
```

Collapsed, this will look like this. The body can be expanded with a click on the header:

![image](https://user-images.githubusercontent.com/37882800/141155383-f09ff1d7-a841-4686-bd9d-aa86e652ab5f.png)

**Comment by storyteller-dev** [comment]
One important notice for the first solution: The page **cannot be within a ForwardOnlySection** or jumping will be disabled and the button will only produce an error message
